core/nuxeo-document.js

```diff
@@ -226,30 +226,58 @@ import './nuxeo-resource.js';
       return path;
     }
 
-    appendClientReason(prop) {
-      prop.viewData = `${prop.data}&clientReason=view`;
-      prop.downloadData = `${prop.data}&clientReason=download`;
+    appendClientReason(documentProp) {
+      let sourceUrl = '';
+      if (documentProp) {
+        if ('url' in documentProp) {
+          sourceUrl = 'url';
+        } else if ('data' in documentProp) {
+          sourceUrl = 'data';
+        }
+      }
+      if (sourceUrl) {
+        documentProp.viewUrl = `${documentProp[sourceUrl]}${
+          documentProp[sourceUrl].indexOf('?') > -1 ? '&clientReason=view' : '?clientReason=view'
+        }`;
+        documentProp.downloadUrl = `${documentProp[sourceUrl]}${
+          documentProp[sourceUrl].indexOf('?') > -1 ? '&clientReason=download' : '?clientReason=download'
+        }`;
+      }
     }
 
     setDocumentViewDownloadProp() {
       const documentProps = [];
-      if (this.documentData && this.documentData.properties) {
-        documentProps.push(
-          this.documentData.properties['file:content'],
-          this.documentData.properties['files:files'],
-          this.documentData.properties['picture:views'],
-          this.documentData.properties['vid:transcodedVideos'],
-          this.documentData.properties['vid:storyboard'],
-        );
+      if (this.documentData) {
+        if (this.documentData.contextParameters) {
+          const documentContextParams = this.documentData.contextParameters;
+          documentProps.push(documentContextParams.preview);
+          documentProps.push(documentContextParams.renditions);
+        }
+
+        if (this.documentData.properties) {
+          const documentDataProperties = this.documentData.properties;
+          documentProps.push(
+            documentDataProperties['file:content'],
+            documentDataProperties['files:files'],
+            documentDataProperties['picture:views'],
+            documentDataProperties['vid:transcodedVideos'],
+            documentDataProperties['vid:storyboard'],
+          );
+        }
+
         documentProps.forEach((docProp) => {
-          if (docProp) {
-            if (Array.isArray(docProp)) {
-              docProp.forEach((prop) => {
-                this.appendClientReason(prop.file ? prop.file : prop.content);
-              });
-            } else {
-              this.appendClientReason(docProp);
-            }
+          if (Array.isArray(docProp)) {
+            docProp.forEach((prop) => {
+              if (prop.file) {
+                this.appendClientReason(prop.file);
+              } else if (prop.content) {
+                this.appendClientReason(prop.content);
+              } else {
+                this.appendClientReason(prop);
+              }
```

Comment:
Kindly optimize above code to below. Remove repetitions.

this.appendClientReason(prop.file?prop.file:(prop.content?prop.content:prop))

OR

this.appendClientReason(prop.file ?? (prop.content ?? prop))

Reply:
both the syntax result in lint issue. so, reverting to this style

Reply:
Kindly discuss. this syntax have repetitive code.